examples_test.go

```diff
@@ -52,3 +52,61 @@ func ExampleConsumer() {
 		}
 	}
 }
+
+// This example shows how to use the consumer to read messages
+// through individual partitions.
+func ExampleConsumer_Partitions() {
+
+	// init (custom) config, enable errors and notifications
+	// set mode to ConsumerModePartitions
+	config := cluster.NewConfig()
+	config.Consumer.Return.Errors = true
+	config.Group.Return.Notifications = true
+	config.Group.Mode = cluster.ConsumerModePartitions
+
+	// init consumer
+	brokers := []string{"127.0.0.1:9092"}
+	topics := []string{"my_topic", "other_topic"}
+	consumer, err := cluster.NewConsumer(brokers, "my-consumer-group", topics, config)
+	if err != nil {
+		panic(err)
+	}
+	defer consumer.Close()
+
+	// trap SIGINT to trigger a shutdown.
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt)
+
+	// consume partitions, watch notifications
+	for {
+		select {
+		case part, more := <-consumer.Partitions():
+			if !more {
+				return
+			}
+
+			// start a separate goroutine to consume the partition;
+			go func(pc cluster.PartitionConsumer) {
+				select {
+				case msg, more := <-pc.Messages():
+					if !more { // exit when the partition is closed
+						return
+					}
+					fmt.Fprintf(os.Stdout, "%s/%d/%d\t%s\t%s\n", msg.Topic, msg.Partition, msg.Offset, msg.Key, msg.Value)
+					consumer.MarkOffset(msg, "") // mark message as processed
+				case err, more := <-consumer.Errors():
```

Comment:
Here you have N instances of selecting over consumer.Errors which won't break, but is not really necessary.. and is not semantically correct as these errors are not partition-specific.

I'd move this outside here to alongside `consumer.Notifications`

This also has the benefit of letting the Messages consumer be a simple `for range` loop

Reply:
so this is one I think I was wondering about, should I expose individual Errors channels or should I continue to multiplex them? it doesn't seem like this is performance critical and it's certainly more convenient to consume them from a single source.